### PR TITLE
fix: add sourcemap file to chunk.auxiliaryFiles

### PR DESCRIPTION
**Pull request by hulin32**
## Summary
fix: add sourcemap file to chunk.auxiliaryFiles (#3253), follow up for #5490

## Test Plan
I created a demo to test it
https://github.com/hulin32/rspack-for-react-test-verification
## Require Documentation?

- [x] No
- [ ] Yes, the corresponding rspack-website PR is \_\_

### CODE REVIEW

File: crates/rspack_plugin_devtool/src/lib.rs [hunk 1]
```diff
@@ -570,6 +570,11 @@ async fn process_assets(&self, compilation: &mut Compilation) -> Result<()> {
     compilation.emit_asset(source_filename.to_owned(), source_asset.clone());
     if let Some((source_map_filename, source_map_asset)) = source_map {
       compilation.emit_asset(source_map_filename.to_owned(), source_map_asset.clone());
+      compilation.chunk_by_ukey.values_mut().for_each(|v| {
+        if v.files.contains(&source_filename) {
+          v.auxiliary_files.insert(source_map_filename.to_owned());
+        }
+      });
```
Comment (SyMind) [inline crates/rspack_plugin_devtool/src/lib.rs:577]: Inserting `auxiliary_files` here is not a good choice. It is better to add them before the following code snippet:

```
mapped_asstes.push(MappedAsset {
    asset: (filename, asset.clone()),
    source_map: Some((source_map_filename, source_map_asset)),
});
```

Additionally, you should remove the following TODO comments:

```rust
// TODO
// chunk.auxiliary_files.add(filename);
```

**Comment by SyMind** [comment]
Please create a unit test within the directory `packages/rspack/tests/cases/chunks`.

File: crates/rspack_plugin_devtool/src/lib.rs (hunk 1 above)
Comment (hulin32) [reply crates/rspack_plugin_devtool/src/lib.rs:577]: thx a lot for your suggestion, will look into it

Comment (hulin32) [reply crates/rspack_plugin_devtool/src/lib.rs:577]: Hey @SyMind, Could you give me some guidance?

I checked the codes, and I found the borrowing rule in that method is not allowing me to change auxiliary_files, do you have any ideas? the only way I can see right now is to change `auxiliary_files` definition to `RefCell<HashSet<String>>`, but  it also will change a lot of places, not sure if it's the good direction

Comment (SyMind) [reply crates/rspack_plugin_devtool/src/lib.rs:577]: I haven't considered it thoroughly. let's stick with your revisions.

**Comment by SyMind** [comment]
!bench

**Comment by SyMind** [comment]
!bench

**Comment by SyMind** [comment]
I ran the performance test and it looks like there are no issues. Then solve the other TODOs and we can be merged.

**Comment by hulin32** [comment]
> Please create a unit test within the directory `packages/rspack/tests/cases/chunks`.

@SyMind dumb question, I wrote the test, but how can I run the test in this folder to test it on my local ?
I am trying with this from [doc](https://web-infra-dev.github.io/rspack-dev-guide/testing/intro.html):
```shell
# In root path
./x build -a
./x test js
```
but not work, thanks

**Comment by SyMind** [comment]
@hulin32 The document is outdated. Please execute the `pnpm run test` command on `packages/rspack` now to perform the test. We will update the document later.

**Comment by hulin32** [comment]
@SyMind Hey, sorry for bother you,  just for local test right now, I am putting below test codes in `Errors.test.js` file, and I want to use imported `webpack`  to compile the file, then check the `stats` to see if `auxiliaryFiles` are good, but seems it isn't using the rspack from source codes, I put a  print log `println!` in `process_assets` method which from `crates/rspack_plugin_devtool/src/lib.rs`, and ran `cargo build && pnpm run build:cli:debug` in root to compile it, then go to `packages/rspack` to run `pnpm run test tests/Errors.test.js`, but the `println!` log is not shown, so for me seems its not using the local one, any suggestions? thx a lot

```js
it("should add source map file to auxiliaryFiles", async () => {
const stats = await new Promise((resolve, reject) => {
  const compiler = webpack({
    entry: {
		main: path.resolve(__dirname, 'cases/chunks/auxiliary-files/one.js')
	},
    devtool: 'source-map',
  });
  try {
    compiler.run((bailedError, stats) => {
      if (bailedError) {
        return reject(bailedError);
      }
      compiler.close(closeError => {
        if (closeError) {
          return reject(closeError);
        }
        resolve(stats);
      });
    });
  } catch (err) {
    // capture sync thrown errors
    reject(err);
  }
});
console.log("statssss", stats.toJson({ errorDetails: false }));
});
```

**Comment by SyMind** [comment]
@hulin32 It's possible that `stats.errors` contains some error messages that could help diagnose the problem.

**Comment by SyMind** [comment]
This is our current issue. Contributing a test case is quite difficult, so I will add the relevant unit tests.

**Review by SyMind — APPROVED** [review]

**Comment by hulin32** [comment]
@SyMind awesome 👍, thanks for your help, also learn something here

**Pull request merged**